jscomp/ext/hash_set.ml

```diff
@@ -42,118 +42,92 @@ let key_index (h : _ Hash_set_gen.t) key =
 
 
 let remove (h : _ Hash_set_gen.t ) key =
```

Comment:
this is painful to sync up different types, lets make `bspp` a subset of  `cpp` to remove such duplication..